janitor/timeseries.py

```diff
@@ -0,0 +1,128 @@
+"""
+Time series-specific data testing and cleaning functions.
+"""
+
+import pandas as pd
+import pandas_flavor as pf
+from pandas.api.types import is_datetime64_any_dtype as is_datetime
+from collections import namedtuple
+from janitor import check
+
+
+def _flag_missing_timestamps(
+    df: pd.DataFrame,
+    frequency: str,
+    column_name: str,
+    first_time_stamp: pd.Timestamp,
+    last_time_stamp: pd.Timestamp,
+) -> namedtuple:
+    """
+    Test if timestamps are missing
+
+    Utility function to test if input data frame is missing any timestamps relative to expected timestamps.
+    They are generated based on the first_time_stamp, last_time_stamp and frequency.
+
+    :param df: data frame which needs to be tested for missing timestamps
+    :param frequency: frequency i.e. sampling frequency of the data.
+        A list of acceptable frequency strings are available here:
+        (https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases)
+    :param column_name: name of the column which has time series if not the index.
+    :param first_time_stamp: timestamp at which the time_series is expected to start from.
+    :param last_time_stamp: timestamp at which the time_series is expected to end with.
+    :return: namedtuple with 3 attributes namely flag, raw_data and new_index
+        1. flag - boolean set to True if there are missing timestamps, else set to False
+        2. raw_data - input data frame as is without any modifications
+        3. new_index - pd.DateTimeIndex that can be used to set the new index.
+            Defaults to None.
+            Assigned a value only when flag is set to True
+    """
+    # Declare a named tuple to hold results
+    MissingTimeStampFlag = namedtuple(
+        "MissingTimeStampFlag", ["flag", "raw_data", "new_index"]
+    )
+    result = {"flag": None, "raw_data": df.copy(deep=True), "new_index": None}
+
+    # Generate expected timestamps
+    expected_timestamps = pd.date_range(
+        start=first_time_stamp, end=last_time_stamp, frequency=frequency
+    )
+
+    # Get actual timestamps
+    if column_name:
+        df = df.set_index(column_name, inplace=True)
```

Comment:
The `inplace=True` was not necessary. This is because adding `inplace=True` returns nothing, so the assignment of `df = df.set_index(..., inplace=True)` would make `df` become a `None` object.

```suggestion
        df = df.set_index(column_name)
```